### DISCUSSION TITLE
Any way to access rate header information?

### DISCUSSION THREAD

**Discussion by 0x4e49434f4c45** [discord-jda/JDA#2369] [Questions and Help] [answered]
Hi!

I am trying to update a channel topic at as regular of an interval as Discord will allow. The topic needs to be updated with the most current data, so for this use case, I can't use JDA's default rate limit handling strategy. For example, if Discord says to retry in 60 seconds, then I'd like to be able to retry in 60 seconds with an updated request, not a replay of the same one.

Additionally, I'm hoping to be able to get the information in the `X-RateLimit-*` headers so that I can pace requests accordingly and not get rate limit errors in the first place.

Is there any way to do this without manually constructing HTTP requests?

**Comment by MinnDevelopment** [comment, marked as answer]
> so that I can pace requests accordingly and not get rate limit errors in the first place.

JDA already does this automatically, it rarely hits rate-limits (unless they are unavoidable). If it *does* hit the rate-limit, it [tells you in the logs](https://github.com/DV8FromTheWorld/JDA/blob/8d583fb6c108ed4deec17f5ff93afea459171318/src/main/java/net/dv8tion/jda/internal/requests/ratelimit/BotRateLimiter.java#L253-L284) and also tells you how long it will wait for.

You can see the rate-limit details in the [trace logs for RateLimiter](https://github.com/DV8FromTheWorld/JDA/blob/8d583fb6c108ed4deec17f5ff93afea459171318/src/main/java/net/dv8tion/jda/internal/requests/ratelimit/BotRateLimiter.java#L302). However, your described situation is considered API abuse and will likely get you API banned for spamming. The rate-limit for updating channel topics and names is 10 minutes, and I would advise you not to try and spam it with updates.

**Reply by 0x4e49434f4c45** [reply]
Thanks for your answer!

> JDA already does this automatically

Just to clarify, I'm aware of this and it's great for when I have actions that can be queued and then sent at the rate Discord allows. I was looking for a programmatic way to obtain the time at which another request will be allowed in order to get the latest data at that time. It sounds like JDA indeed doesn't expose this.
